internetarchive/session.py

```diff
@@ -402,19 +413,19 @@ def submit_task(self, identifier, cmd, comment=None, priority=None, data=None,
 
         :rtype: :class:`requests.Response`
         """
-        headers = {} if not headers else headers
-        if reduced_priority is not None:
+        headers = headers or {}
+        if reduced_priority:
             headers.update({'X-Accept-Reduced-Priority': '1'})
-
-        c = Catalog(self, request_kwargs)
-        r = c.submit_task(identifier, cmd,
-                          comment=comment,
-                          priority=priority,
-                          data=data,
-                          headers=headers)
-        return r
-
-    def iter_history(self, identifier, params=None, request_kwargs=None):
+        return Catalog(self, request_kwargs).submit_task(identifier, cmd,
+                                                         comment=comment,
+                                                         priority=priority,
+                                                         data=data,
+                                                         headers=headers)
+
+    def iter_history(self,
+                     identifier: str | None = None,
```

Comment:
Hmm, so `identifier` was already optional per the docstring, just not per the function signature. Explicitly passing `None` was already supported (since it essentially gets passed to `requests.get`, and [`None` values get stripped](https://docs.python-requests.org/en/latest/user/quickstart/#passing-parameters-in-urls)), but this change does mean that the method now requests the entire task history by default. I have no particular opinion on whether or not that's a good idea.

Reply:
Good catch.  Thanks.  Fixing...